scripts/uosc/elements/Menu.lua

```diff
@@ -255,21 +273,26 @@ function Menu:update_content_dimensions()
 end
 
 function Menu:update_dimensions()
-	-- Coordinates and sizes are of the scrollable area to make
-	-- consuming values in rendering and collisions easier. Title is rendered
+	-- Coordinates and sizes are of the scrollable area. Title is rendered
 	-- above it, so we need to account for that in max_height and ay position.
+	-- This is a debt from an era where we had different cursor event handling,
+	-- and dumb titles with no search inputs. It could use a refactor.
 	local min_width = state.fullormaxed and options.menu_min_width_fullscreen or options.menu_min_width
-	local height_available = display.height - Elements.timeline.size_min
+	local vertical_padding = round(self.scroll_step / 2)
+	local height_available = display.height - vertical_padding * 2
 
 	for _, menu in ipairs(self.all) do
 		local width = math.max(menu.search and menu.search.width or 0, menu.max_width)
 		menu.width = round(clamp(min_width, width, display.width * 0.9))
-		local title_height = (menu.is_root and menu.title or menu.search) and self.scroll_step or 0
-		local max_height = round(height_available * 0.9 - title_height)
+		local title_height = (menu.title or menu.search) and self.scroll_step or 0
+		local max_height = height_available - title_height
 		local content_height = self.scroll_step * #menu.items
 		menu.height = math.min(content_height - self.item_spacing, max_height)
 		local search_top = menu.search and menu.search.top or height_available
-		menu.top = math.min(search_top, round((height_available - menu.height + title_height) / 2))
+		menu.top = math.max(
+			title_height + vertical_padding,
+			math.min(search_top, round((height_available - menu.height + title_height) / 2))
+		)
```

Comment:
In the root menu it now properly accounts for the search bar for small windows when it doesn't have a title.
![image](https://github.com/tomasklaen/uosc/assets/8932183/258c20cc-339d-4cd8-bd96-06eed9c6a773)
![image](https://github.com/tomasklaen/uosc/assets/8932183/02c28b20-bfe0-4e84-bb79-3bee4bdb8df6)
But submenus are always moved down a bit because they do have titles and they require the `menu.is_root` check.
![image](https://github.com/tomasklaen/uosc/assets/8932183/f2c9de3b-25ab-4567-86b9-90a19e1314d9)

Reply:
I forgot to mention that since this change is independent of the palette mode it should be a separate commit imo.

Reply:
True, I'll split it into a separate PR.